packages/docs/src/guide/errors.md

```
@@ -2,35 +2,68 @@
 }
 ```
 
-A key distinction:
+Two subclasses narrow the failure mode:
 
-- **HTTP error** (4xx/5xx) → `response` is set, `status` reflects it.
-- **Network error** (DNS failure, connection reset, abort) → `response` is `undefined`, and `code` is set instead.
+- **`HttpResponseError`** — the server answered with a `4xx`/`5xx` status. `response` and `status` are set; `code` defaults to `'HTTP_RESPONSE_ERROR'`.
+- **`NetworkError`** — the request never produced a response (DNS failure, connection reset, abort). `response` is `undefined` and `code` carries the reason (e.g. `'ECONNRESET'`, `'ECONNABORTED'`).
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)